### PR TITLE
Fix tests after adding Options service #217

### PR DESCRIPTION
**Pull request by JunTaoLuo**
@javiercn

### CODE REVIEW

File: test/Microsoft.Extensions.Localization.Tests/LocalizationServiceCollectionExtensionsTest.cs [hunk 1]
```diff
@@ -21,15 +21,23 @@ public void AddLocalization_AddsNeededServices()
 
             // Assert
             var services = collection.ToList();
-            Assert.Equal(2, services.Count);
+            Assert.Equal(4, services.Count);
```
Comment (pranavkm) [inline test/Microsoft.Extensions.Localization.Tests/LocalizationServiceCollectionExtensionsTest.cs]:
```C#
Assert.Collection(services,
    service => 
    {
        Assert.Equal(typeof(IOptions<>), service.ServiceType);
        Assert.Equal(typeof(OptionsManager<>), service.ImplementationType);
        Assert.Equal(ServiceLifetime.Singleton, service.Lifetime);
    },
    service =>
    {
        Assert.Equal(typeof(IOptionsMonitor<>), service.ServiceType);
        Assert.Equal(typeof(OptionsMonitor<>), service.ImplementationType);
        Assert.Equal(ServiceLifetime.Singleton, service.Lifetime);
    },
    service =>
    {
        Assert.Equal(typeof(IStringLocalizerFactory), service.ServiceType);
        Assert.Equal(typeof(ResourceManagerStringLocalizerFactory), service.ImplementationType);
        Assert.Equal(ServiceLifetime.Singleton, service.Lifetime);
    },
    service =>
    {
        Assert.Equal(typeof(IStringLocalizer<>), service.ServiceType);
            Assert.Equal(typeof(StringLocalizer<>), service.ImplementationType);
            Assert.Equal(ServiceLifetime.Transient, service.Lifetime);
    });
```

Comment (pranavkm) [inline test/Microsoft.Extensions.Localization.Tests/LocalizationServiceCollectionExtensionsTest.cs]: Makes it easier if the order ever changes.

File: test/Microsoft.Extensions.Localization.Tests/LocalizationServiceCollectionExtensionsTest.cs [hunk 2]
```diff
@@ -21,15 +21,23 @@ public void AddLocalization_AddsNeededServices()
...
 
-            Assert.Equal(typeof(IStringLocalizerFactory), services[0].ServiceType);
-            Assert.Equal(typeof(ResourceManagerStringLocalizerFactory), services[0].ImplementationType);
+            Assert.Equal(typeof(IOptions<>), services[0].ServiceType);
```
Comment (kichalla) [inline test/Microsoft.Extensions.Localization.Tests/LocalizationServiceCollectionExtensionsTest.cs]: Ideally I think we do not want to test the services added by options itself as it makes test easy to break if Options decides to change the services (ex: adds/deletes a service etc.). I think we should just check for the services that localization adds itself. (ex: https://github.com/aspnet/Localization/pull/221)

Comment (JunTaoLuo) [inline test/Microsoft.Extensions.Localization.Tests/LocalizationServiceCollectionExtensionsTest.cs]: Is Options a required service for Localization though? If it is required to be there, we should make sure it's added when `AddLocalization` is called no? @javiercn?

**Comment by JunTaoLuo** [comment]
#221 will be the approach we take. Closing

**Pull request closed without merging**